src/Service/BucketService.php

```diff
@@ -0,0 +1,39 @@
+<?php
+
+namespace App\Service;
+
+use League\Flysystem\Filesystem;
+
+class BucketService
+{
+    private $filesystem;
+
+    public function __construct(Filesystem $filesystem) {
+        $this->filesystem = $filesystem;
+    }
+
+    public function uploadFile ($file, $dir) {
+        $filesystem = $this->filesystem;
+        $path = "/$dir/".$file->getClientOriginalName();
+
+        $stream = fopen($file->getRealPath(), 'r+');
+        $filesystem->writeStream($path, $stream, ['visibility' => 'public']);
+        fclose($stream);
+
+        return $path;
+    }
+
+    public function retrieveFile ($path) {
+        $filesystem = $this->filesystem;
+
+        $stream = $filesystem->readStream($path);
+        $contents = stream_get_contents($stream);
```

Comment:
Are the `$contents` supposed to be used somewhere?

Reply:
Actually this function is not needed since we can access to the file directly with the cloudfront domain